integration-tests/load-tests/load-test-helpers.ts

```diff
@@ -0,0 +1,226 @@
+import { ethers } from 'ethers'
+import { Contract } from '@ethersproject/contracts'
+import { OptimismEnv } from '../test/shared/env'
+import { Direction } from '../test/shared/watcher-utils'
+
+interface TransactionParams {
+  contract: Contract
+  functionName: string
+  functionParams: any[]
+}
+
+export const executeL1ToL2Transactions = async (
+  env: OptimismEnv,
+  txs: TransactionParams[]
+) => {
+  for (const tx of txs) {
+    const signer = ethers.Wallet.createRandom().connect(env.l1Wallet.provider)
+    const receipt = await env.l1Messenger
+      .connect(signer)
+      .sendMessage(
+        tx.contract.address,
+        tx.contract.interface.encodeFunctionData(
+          tx.functionName,
+          tx.functionParams
+        ),
+        8000000,
+        {
+          gasPrice: 0,
+        }
+      )
+
+    await env.waitForXDomainTransaction(receipt, Direction.L1ToL2)
```

Comment:
Instead of waiting inside the loop, can maybe wait outside the loop so that all L1 transactions get submitted w/o blocking?

Reply:
I wanted this function to be blocking, the parallel version is not blocking.